zhaquirks/sonoff/snzb04p.py

```diff
@@ -0,0 +1,64 @@
+"""Sonoff SNZB-04 device."""
+
+from zigpy import types
+from zigpy.quirks import CustomCluster
+from zigpy.quirks.v2 import (
+    BinarySensorDeviceClass,
+    ClusterType,
+    EntityType,
+    QuirkBuilder,
+)
+from zigpy.zcl.clusters.general import OnOff
+from zigpy.zcl.foundation import BaseAttributeDefs, ZCLAttributeDef
+
+
+class SonoffOnOffCluster(CustomCluster, OnOff):
+    """Sonoff OnOff cluster for contact sensor."""
+
+    """Prevents the creation of the on_off entity since it is not functional."""
+
+    cluster_id = 6  # 0x0006
+    name = "Sonoff OnOff cluster"
+    ep_attribute = "sonoff_onoff_cluster"
+    SKIP_CONFIGURATION = True
```

Comment:
Similar to the Hue sensor PR, let's remove this cluster for now and not modify it.
But you can already add the following to the v2 quirk (from https://github.com/zigpy/zigpy/pull/1535/).
```python
.prevent_default_entity_creation(endpoint_id=1, cluster_id=OnOff.cluster_id)
```

Unfortunately, it's not implemented in ZHA yet (blocked by https://github.com/zigpy/zha/pull/365), but it will be in the future. This will be the proper way to disable ZHA's default entities.